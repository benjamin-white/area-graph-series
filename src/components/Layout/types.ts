export type LayoutProps = {
  children: React.ReactNode
  className?: string | string[]
}
