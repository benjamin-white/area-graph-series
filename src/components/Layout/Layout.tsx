import { LayoutProps } from './types'
import clsx from 'clsx'

const Layout = ({ children, className }: LayoutProps) => (
  <div className={clsx('site-wrap', className)}>
    <div className="container">{children}</div>
  </div>
)

export default Layout
