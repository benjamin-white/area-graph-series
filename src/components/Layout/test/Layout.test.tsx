import { render, screen } from '@testing-library/react'
import Layout from '../'

describe('Layout', () => {
  it('renders a Layout component', () => {
    const { container } = render(<Layout>Lorem ipsum</Layout>)

    expect(screen.getByText('Lorem ipsum')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('accepts a className prop', () => {
    const { container } = render(
      <Layout className="class-name">Lorem ipsum</Layout>,
    )

    expect(container.firstChild).toHaveClass('class-name')
  })
})
