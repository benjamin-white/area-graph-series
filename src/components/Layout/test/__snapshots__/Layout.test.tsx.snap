// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > renders a Layout component 1`] = `
<div
  class="site-wrap"
>
  <div
    class="container"
  >
    Lorem ipsum
  </div>
</div>
`;
